doc/source/development/rfc/rfc104_gdal_cli.rst

```diff
@@ -0,0 +1,464 @@
+
+In this automated mode, no switch besides open options can be specified, given that we don't know yet in which mode to open.
+
+If the dataset has both raster and vector content, an error will be emitted, inviting the user to specify explicitly the raster or vector mode.
+
+Example:
+
+  .. code-block:: shell
+
+        gdal info my.tif
+
+        gdal info my.gpkg
+
+gdal info raster
+++++++++++++++++
+
+Equivalent of existing :ref:`gdalinfo`
+
+Synopsis: ``gdal info raster [-i <filename>] [other options] <filename>``
+
+Example:
+
+  .. code-block:: shell
+
+        gdal info raster my.gpkg
+
+Switches:
+
+* ``-f json|text``, ``--of json|text``: output format. Will default to JSON.
+
+* ``--min-max``
+
+* ``--stats``
+
+* ``--approx-stats``
+
+* ``--hist``
+
+* ``--no-gcp``
+
+* ``--no-md``
+
+* ``--no-ct``
+
+* ``--no-fl``
+
+* ``--no-nodata``
+
+* ``--no-mask``
+
+* ``--checksum``
+
+* ``--list-mdd``
+
+* ``--mdd <domain>|all``
+
+* ``--subdataset <num>``
+
+gdal info vector
+++++++++++++++++
+
+Equivalent of existing :ref:`ogrinfo`
+
+Synopsis: ``gdal info vector [-i <filename>] [other options] <filename> [<layername>]...``
+
+Example:
+
+  .. code-block:: shell
+
+        gdal info vector my.gpkg
+
+Switches:
+
+* ``-f json|text``, ``--of json|text``: output format. Will default to JSON.
+
+* ``--sql <statement>``
+
+* ``-l <name>``, ``--layer <name>``
+
+* ``--update``: New default will be read-only
+
+* ``--interleaved-layers``: a.k.a random layer reading mode (ogrinfo ``-al``), for OSM and GMLAS mostly.
+
+* ``--where <statement>``
+
+* ``--dialect <dialectname>``
+
+* ``--bbox <xmin> <ymin> <xmax> <ymax>``
+
+TODO
+
+gdal info multidim
++++++++++++++++++++
+
+Equivalent of existing :ref:`gdalmdiminfo`
+
+* ``-f json``, ``--of json``: output format. Only JSON, as there is currently no text based format.
+
+TODO
+
+
+gdal convert raster
++++++++++++++++++++
+
+Equivalent of existing :ref:`gdal_translate`
+
+TODO
+
+gdal convert vector
++++++++++++++++++++
+
+Equivalent of existing :ref:`ogr2ogr`
+
+TODO
+
+gdal convert multidim
++++++++++++++++++++++
+
+Equivalent of existing :ref:`gdalmdimtranslate`
+
+TODO
+
+gdal warp
++++++++++
+
+.. note::
+
+    In the User Survey, a number of users have expressed a wish to have
+    gdal_translate and gdalwarp functionality merged together. This RFC does not
+    attempt at addressing that. Or should it... ? That'd be a huge topic
+
+TODO: warp is also a bit of a misnomer as gdalwarp can mosaic.
+
+Equivalent of existing :ref:`gdalwarp`
+
+TODO
+
+gdal contour
+++++++++++++
+
+Equivalent of existing :ref:`gdal_contour`
+
+TODO
+
+gdal rasterize
+++++++++++++++
+
+Equivalent of existing :ref:`gdal_rasterize`
+
+TODO
+
+gdal create
++++++++++++
+
+(or ``gdal create raster`` in case we'd have a vector creation one one day?
+
+TODO
+
+gdal footprint
+++++++++++++++
+
+Equivalent of existing :ref:`gdal_footprint`
+
+TODO
+
+gdal viewshed
```

Comment:
I wonder if it makes sense for "viewshed" to be treated like the other DEM operations and put under `gdal dem`.